.github/workflows/ci.yml

```
@@ -16,217 +16,51 @@ on:
 name: CI
 
 env:
-  toolchain: nightly-2021-11-20
   CARGO_HTTP_MULTIPLEXING: false
   CARGO_TERM_COLOR: always
   PROTOC: protoc
-  TERM: unkown
-  
+
 jobs:
-  clippy:
-    name: clippy
-    runs-on: [ self-hosted, ubuntu18.04-high-cpu ]
-    steps:
-      - name: checkout
-        uses: actions/checkout@v2
-      - name: toolchain
-        uses: actions-rs/toolchain@v1
-        with:
-          toolchain: ${{ env.toolchain }}
-          components: clippy, rustfmt
-          override: true
-      - name: ubuntu dependencies
-        run: |
-          sudo apt-get update && \
-          sudo apt-get -y install \
-          build-essential \
-          libgtk-3-dev \
-          libwebkit2gtk-4.0-dev \
-          libsoup2.4-dev \
-          curl \
-          wget \
-          libappindicator3-dev \
-          patchelf \
-          librsvg2-dev \
-          libprotobuf-dev \
-          protobuf-compiler
-      - name: cargo fmt
-        uses: actions-rs/cargo@v1
-        with:
-          command: fmt
-          args: --all -- --check
-      - name: Install cargo-lints
-        uses: actions-rs/cargo@v1
-        with:
-          command: install
-          args: cargo-lints
-      - name: Clippy check (with lints)
-        uses: actions-rs/cargo@v1
-        with:
-          command: lints
-          args: clippy --all-targets --all-features
-  build:
-    name: check nightly
+  check-nightly:
     runs-on: [ self-hosted, ubuntu18.04-high-cpu ]
-    steps:
-      - name: checkout
-        uses: actions/checkout@v2
-      - name: toolchain
-        uses: actions-rs/toolchain@v1
-        with:
-          toolchain: ${{ env.toolchain }}
-          profile: minimal
-          override: true
-      - name: ubuntu dependencies
-        run: |
-          sudo apt-get update && \
-          sudo apt-get -y install \
-          build-essential \
-          libgtk-3-dev \
-          libwebkit2gtk-4.0-dev \
-          libsoup2.4-dev \
-          curl \
-          wget \
-          libappindicator3-dev \
-          patchelf \
-          librsvg2-dev \
-          libprotobuf-dev \
-          protobuf-compiler
-      - name: cargo check
-        uses: actions-rs/cargo@v1
-        with:
-          command: check
-          args: --release --all-targets
-      - name: cargo check ffi separately
-        uses: actions-rs/cargo@v1
-        with:
-          command: check
-          args: --release --package tari_wallet_ffi
+    uses:
+      ./.github/workflows/reusable-ci-jobs.yml
+    with:
+      check-ffi: true
+
+  check-stable:
+    uses:
```

Not sure if this works, but we need this to run the jobs on self-hosted for better run times
```suggestion
    runs-on: [ self-hosted, ubuntu18.04-high-cpu ]
    uses:
```

---

And same for other jobs

---

Probably, but I left everything as they're currently configured.